snapshot_tests/shader/base_params_output_resolution.wgsl

```diff
@@ -0,0 +1,40 @@
+struct VertexInput {
+    @location(0) position: vec3<f32>,
+    @location(1) tex_coords: vec2<f32>,
+}
+
+struct VertexOutput {
+    @builtin(position) position: vec4<f32>,
+    @location(0) tex_coords: vec2<f32>,
+}
+
+@group(0) @binding(0) var textures: binding_array<texture_2d<f32>, 16>;
+@group(2) @binding(0) var sampler_: sampler;
+
+struct BaseShaderParameters {
+    plane_id: i32,
+    time: f32,
+    output_resolution: vec2<u32>,
+    texture_count: u32,
+}
+
+var<push_constant> base_params: BaseShaderParameters;
+
+@vertex
+fn vs_main(input: VertexInput) -> VertexOutput {
+    var output: VertexOutput;
+
+    let scale_factor_x = f32(base_params.output_resolution.x) / 4000.0;
+    let scale_factor_y = f32(base_params.output_resolution.y) / 2000.0;
+
+    output.position = vec4(input.position.x * scale_factor_x, input.position.y * scale_factor_y, input.position.z, 1.0);
```

Comment:
that feels very synthetic, I'm not sure based on this code what would be correct output